### PR TITLE
PHOENIX-6066 : MetaDataEndpointImpl.doGetTable should acquire a readLock instead of an exclusive writeLock 

### PR DESCRIPTION
**Pull request by palashc**

### CODE REVIEW

File: phoenix-core-client/src/main/java/org/apache/phoenix/query/QueryServices.java [hunk 1]
```diff
@@ -486,6 +486,8 @@ public interface QueryServices extends SQLCloseable {
 
     int DEFAULT_PHOENIX_GET_REGIONS_RETRIES = 10;
 
+    String PHOENIX_GET_METADATA_READ_LOCK_ENABLED = "phoenix.get.metadata.read.lock.enabled";
```
Comment (dbwong) [inline phoenix-core-client/src/main/java/org/apache/phoenix/query/QueryServices.java:489]: Whats the risk that requires this be parameterized?  Also when did the previous change with exclusive writelock go ou?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/MetadataGetTableReadLockIT.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ */
+@Category(ParallelStatsDisabledTest.class)
+public class MetadataGetTableReadLockIT extends BaseTest {
+
+    @BeforeClass
+    public static synchronized void doSetup() throws Exception {
+        Map<String, String> props = Maps.newHashMapWithExpectedSize(1);
+        // Disable system task handling
+        props.put(QueryServices.TASK_HANDLING_INITIAL_DELAY_MS_ATTRIB, Long.toString(Long.MAX_VALUE));
+        props.put(QueryServices.PHOENIX_GET_METADATA_READ_LOCK_ENABLED, Boolean.toString(true));
```
Comment (dbwong) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/MetadataGetTableReadLockIT.java:35]: Technically don't have either an integration/unit test that checks that when this is false there is no change.

**Review by dbwong — COMMENTED** [review]
Generally good, couple of questions.

File: phoenix-core-client/src/main/java/org/apache/phoenix/query/QueryServices.java (hunk 1 above)
Comment (palashc) [reply phoenix-core-client/src/main/java/org/apache/phoenix/query/QueryServices.java:489]: Just the ability to quickly to disable this change in production if it causes any issues.

Comment (dbwong) [reply phoenix-core-client/src/main/java/org/apache/phoenix/query/QueryServices.java:489]: Is it not enabled prior to this release?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/MetadataGetTableReadLockIT.java (hunk 2 above)
Comment (palashc) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/MetadataGetTableReadLockIT.java]: We decided to keep the config enabled by default.

File: phoenix-core-client/src/main/java/org/apache/phoenix/query/QueryServices.java (hunk 1 above)
Comment (palashc) [reply phoenix-core-client/src/main/java/org/apache/phoenix/query/QueryServices.java:489]: > Also when did the previous change with exclusive write lock go out?
@dbwong This has always been the behaviour in production. We want to change it to using read locks for getTable RPC calls.

**Pull request closed without merging**